plugins/org.polarsys.capella.scenario.editor/src/org/polarsys/capella/scenario/editor/helper/EmbeddedEditorInstanceHelper.java

```diff
@@ -102,9 +107,13 @@ public static EObject getScenarioLevel() {
   public static List<String> getExchangeNames(String source, String target) {
     List<String> messages = new ArrayList<String>();
 
-    List<AbstractEvent> exchanges = getAvailableExchanges(source, target);
-    for (AbstractEvent exchange : exchanges) {
-      messages.add(exchange.getName());
+    List<AbstractEventOperation> exchanges = getAvailableExchanges(source, target);
+    for (AbstractEventOperation exchange : exchanges) {
+      if (isInterfaceScenario()) {
```

Comment:
Cred ca se poate scoate isInterfaceScenario() inaintea for-ului, pare ca nu depinde de exchange